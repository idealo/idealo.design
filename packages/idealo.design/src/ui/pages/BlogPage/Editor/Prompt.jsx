import React from "react";
import ReactModal from "react-modal";

export default function PromptSuccess(props) {
  return (
      <ReactModal
          isOpen={props.show}
          style={{
            overlay: {
              backgroundColor: 'rgba( 190, 190, 190, 0.75)'
            },
            content: {
              top: '40%',
              left: '25%',
              right: '25%',
              bottom: '40%',
            }
          }}
      >
          <header>
              <p>{props.message}</p>
          </header>
          <footer>
              {props.message === 'Do you want to delete or archive that post?'
                  ?<div>
                      <button onClick={props.onCancel}>Close</button>
                      <button onClick={props.onArchive}>Archive</button>
                      <button onClick={props.onDelete} title='promptDeleteButton'>Delete</button>
                  </div>:
                  <div>
                      <button onClick={props.onHide}>Close</button>
                      <button onClick={props.onLeave}>Yes</button>
                  </div>
              }
          </footer>

      </ReactModal>
  );
}
