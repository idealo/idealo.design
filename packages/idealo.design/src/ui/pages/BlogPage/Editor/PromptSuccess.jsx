import React from "react";
import {Modal } from "react-bootstrap";

export default function PromptSuccess(props) {
    const { message, onLeave, ...rest } = props;

    return (
        <Modal
            {...rest}
            size="md"
            aria-labelledby="contained-modal-title-vcenter"
            centered
        >
            <Modal.Header submitButton>
                <Modal.Title id="contained-modal-title-vcenter">
                </Modal.Title>
            </Modal.Header>
            <Modal.Body>
                <p>{'Dein Blogpost wurde erfolglreich gespeichert'}</p>
            </Modal.Body>
        </Modal>
    );
}
