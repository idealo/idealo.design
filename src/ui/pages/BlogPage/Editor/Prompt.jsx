import React from "react";
import ReactModal from "react-modal";
import s from "./Editor.module.scss";

export default function PromptSuccess(props) {
    return (
        <ReactModal
            isOpen={props.show}
            style={{
                overlay: {
                    backgroundColor: "rgba( 190, 190, 190, 0.75)",
                },
                content: {
                    top: "40%",
                    left: "30%",
                    right: "30%",
                    bottom: "45%",
                },
            }}
        >
            <header>
                <h2 className={s.promptMessage}>{props.message}</h2>
            </header>
            <footer>
                {props.message === "Do you want to delete or archive that post?" ? (
                    <div className={s.buttonDiv}>
                        <button onClick={props.onCancel} className={s.buttons}>Close</button>
                        <button onClick={props.onArchive} className={s.buttons}>Archive</button>
                        <button onClick={props.onDelete} title="promptDeleteButton" className={s.buttons}>Delete
                        </button>
                    </div>
                ) : (
                    <div className={s.buttonDiv}>
                        <button onClick={props.onHide} className={s.buttons}>Close</button>
                        <button onClick={props.onLeave} className={s.buttons}>Yes</button>
                    </div>
                )}
            </footer>
        </ReactModal>
    );
}
