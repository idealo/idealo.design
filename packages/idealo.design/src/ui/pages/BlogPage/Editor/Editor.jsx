import React from 'react'
import ReactModal from 'react-modal';

import { withRouter } from 'react-router'

import { Editor, EditorState, getDefaultKeyBinding, RichUtils, convertToRaw, convertFromRaw } from "draft-js";
import '~/draft-js/dist/Draft.css'
import s from './Editor.module.scss';
import Prompt from './Prompt';
import PromptSuccess from "./PromptSuccess";
import { fetchSinglePost, updateSinglePost, fetchDistinctCategories} from '../data';
import CreatableSelect from 'react-select/creatable';
import slugify from "slugify";

class RichTextEditor extends React.Component {
  constructor(props) {
    super(props);

    const { match, location, history } = props;
    this.blog = null;
    this.slug = match.params.slug;
    this.mode = this.slug ? 'EDIT' : 'CREATE';
    this.state = {
      editorState: EditorState.createEmpty(),
      title: '',
      categoryDisplayValue: '',
      categorySlug: '',
      isPromptOpen: false,
      isEdited: false,
      lastHistoryLocation: '',
      isSubmitPromptOpen: false,
      cats: [],
      error: [],
    };

    this.focus = () => this.refs.editor.focus();
    this.onChange = (editorState) => {
      this.setState({ editorState, isEdited: true });
      this.handleValidation();
    }

    this.handleKeyCommand = this._handleKeyCommand.bind(this);
    this.mapKeyToEditorCommand = this._mapKeyToEditorCommand.bind(this);
    this.toggleBlockType = this._toggleBlockType.bind(this);
    this.toggleInlineStyle = this._toggleInlineStyle.bind(this);
    this.handleCancellation = this.handleCancellation.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onModalCancel = this.onModalCancel.bind(this);
    this.onModalLeave = this.onModalLeave.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleSelectChange = this.handleSelectChange.bind(this);
    this.handleSelectCreate = this.handleSelectCreate.bind(this);
    this.handleValidation = this.handleValidation.bind(this);
  }

  async componentDidMount() {
    this.props.history.block((tx) => {
      if (this.state.isEdited) {
        this.setState({ lastHistoryLocation: tx.pathname, isPromptOpen: true });
        return !this.state.isEdited;
      }
      return true;
    })

    this.setState({ cats: await fetchDistinctCategories() });

    if(this.slug) {
      this.blog = await fetchSinglePost({ slug: this.slug });
      const contentState = convertFromRaw( this.blog.blogpostcontent);

      this.setState({
        blogpost: this.blog, // refactor
        title: this.blog.title,
        categorySlug: this.blog.categoryslug,
        categoryDisplayValue: this.blog.categorydisplayvalue,
        editorState: EditorState.createWithContent(contentState)
      })
    }
    ReactModal.setAppElement('body');
  }

  renderContentAsRawJs() {
    const contentState = this.state.editorState.getCurrentContent();
    const raw = convertToRaw(contentState);

    return JSON.stringify(raw, null, 2);
  }

  _handleKeyCommand(command, editorState) {
    const newState = RichUtils.handleKeyCommand(editorState, command);
    if (newState) {
      this.onChange(newState);
      return true;
    }
    return false;
  }

  _mapKeyToEditorCommand(e) {
    if (e.keyCode === 9 /* TAB */) {
      const newEditorState = RichUtils.onTab(
        e,
        this.state.editorState,
        4, /* maxDepth */
      );
      if (newEditorState !== this.state.editorState) {
        this.onChange(newEditorState);
      }
      return null;
    }

    return getDefaultKeyBinding(e);
  }

  _toggleBlockType(blockType) {
    this.onChange(
      RichUtils.toggleBlockType(
        this.state.editorState,
        blockType
      )
    );
  }

  _toggleInlineStyle(inlineStyle) {
    this.onChange(
      RichUtils.toggleInlineStyle(
        this.state.editorState,
        inlineStyle
      )
    );
  }


  handleCancellation(e) {
    if(this.state.isEdited) {
      this.setState({isPromptOpen: true });
    } else {
      this.props.history.push('/blog');
    }
  }

  handleValidation() {
    let formIsValid = true;
    let errors = {};
    const blacklist = ['new-post'];

    if (!this.state.title) {
      formIsValid = false;
      errors['title'] = 1;
    }

    blacklist.map(word => {
      if (word === slugify(this.state.title).replace(/^\s+|\s+$/g, '').toLowerCase()) {
        formIsValid = false;
        errors ['title'] = 1;
      }
    });

    if (!this.state.categoryDisplayValue) {
      formIsValid = false;
      errors['categoryDisplayValue'] = 1;
    }

    if (!this.state.editorState.getCurrentContent().getPlainText()) {
      formIsValid = false;
      errors['editorState'] = 1;
    }
    this.setState({error: errors });

    return formIsValid;
  }

  handleSubmit(e){
    e.preventDefault();
    if(!this.handleValidation()){
      alert("Form has errors.")
      return;
    }

    if(this.mode === 'EDIT') {
      this.props.history.block(() => true);
      this.blog.title = this.state.title;
      this.blog.blogpostcontent = this.renderContentAsRawJs();
      this.blog.categoryDisplayValue = this.state.categoryDisplayValue;
      this.blog.categorySlug = this.state.categorySlug;
      updateSinglePost({
        slug: this.slug,
        post: this.blog
      }, () => {
        this.setState({ isSubmitPromptOpen: true });
        setTimeout(() => {
          this.setState({ isSubmitPromptOpen: false }, () => {
            this.props.history.push('/blog');
          });
        }, 1500);
      })
    return;
    }

    const body = JSON.stringify({
      title: this.state.title,
      categoryDisplayValue: this.state.categoryDisplayValue,
      categorySlug: this.state.categorySlug,
      blogpostcontent: this.renderContentAsRawJs(),
    })

    fetch('/api/blogposts', {
      method: 'POST',
      headers: { 'Accept': 'application/json', 'Content-Type': 'application/json' },
      body
    }).then(function(response) {
      console.log(response)
      return response.json();
    });
       this.props.history.block(() => {return true;})
      this.setState({ isSubmitPromptOpen: true });
      setTimeout(() => {
        this.setState({ isSubmitPromptOpen: false }, () => {
          this.props.history.push('/blog');
      });
    }, 1500);
    //this.setState({editorState: EditorState.createEmpty()});
  }

  onModalCancel() {
    this.setState({isPromptOpen: false });
  }

  onModalLeave() {
    this.setState(
      { isPromptOpen: false, isEdited: false },
      () => {
        this.props.history.push('/blog');
      });
  }

  handleChange(event) {
    const target = event.target;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const name = target.name;

    if (target.type === 'select-one') {
      const el = this.state.cats.filter(cat => cat.categoryslug === value).pop();
      this.setState({
        categoryDisplayValue: el.categorydisplayvalue
      })
    }

    this.setState({
      [name]: value
    });

    this.handleValidation();
  }

  async handleSelectChange(newValue, actionMeta) {
    await this.setState({
      categorySlug: newValue.categoryslug,
      categoryDisplayValue: newValue.categorydisplayvalue
    })
    this.handleValidation();
  }

  handleSelectCreate(inputValue) {
    const { cats } = this.state;
    const newOption = {
      categoryslug: inputValue.toLowerCase().replace(/\W/g, ''),
      categorydisplayvalue: inputValue
    }
    this.setState({
      cats: [...cats, newOption],
      categorySlug: newOption.categoryslug,
      categoryDisplayValue: newOption.categorydisplayvalue
    });
  }

  render() {
    const {editorState} = this.state;
    // If the user changes block type before entering any text, we can
    // either style the placeholder or hide it. Let's just hide it now.
    let className = s['RichEditor-editor'];
    const contentState = editorState.getCurrentContent();
    if (!contentState.hasText()) {
      if (contentState.getBlockMap().first().getType() !== 'unstyled') {
        className += ' ' + s['RichEditor-hidePlaceholder'];
      }
    }

    return (
      <>
        {this.mode === 'EDIT'
         ? <h1>Edit blogpost</h1>
         : <h1>Create blogpost</h1>}

        <div className={s.InputFields}>
          <input className={this.state.error['title'] ? s.empty : ''} onChange={this.handleChange} name="title" value={this.state.title} placeholder="Titel"/>
          <form name="category" className="select-container">
            <CreatableSelect
                getOptionLabel={option => option.categorydisplayvalue}
                getOptionValue={option => option.categoryslug}
                onChange={this.handleSelectChange}
                onCreateOption={this.handleSelectCreate}
                options={this.state.cats}
                value={{categorydisplayvalue: this.state.categoryDisplayValue, categoryslug: this.state.categorySlug}}
                getNewOptionData={(inputValue, optionLabel) => ({
                  categoryslug: inputValue,
                  categorydisplayvalue: optionLabel,
                  __isNew__: true
                })}
                className={this.state.error['categoryDisplayValue'] ? s.empty : ''}
            />
          </form>
        </div>

        <div className={this.state.error['editorState'] ? s.empty + ' ' + s["RichEditor-root"] : s["RichEditor-root"]}>
           <BlockStyleControls
            editorState={editorState}
            onToggle={this.toggleBlockType}
          />
          <InlineStyleControls
            editorState={editorState}
            onToggle={this.toggleInlineStyle}
          />
          <div onClick={this.focus}>
            <Editor
              blockStyleFn={getBlockStyle}
              customStyleMap={styleMap}
              editorState={editorState}
              handleKeyCommand={this.handleKeyCommand}
              keyBindingFn={this.mapKeyToEditorCommand}
              onChange={this.onChange}
              placeholder="Tell a story..."
              ref="editor"
              spellCheck={true}
            />
          </div>
        </div>
        <div className={s['newBlogPostButtons']}>
          <button className={s['SubmitButton']} onClick={this.handleSubmit}>Submit</button>
          <button className={s['CancelButton']} onClick={this.handleCancellation}>Cancel</button>
        </div>
          <Prompt
              show={this.state.isPromptOpen}
              onHide={this.onModalCancel}
              onLeave={this.onModalLeave}
              message='Are you sure you want to leave?'
          />
          <PromptSuccess
              show={this.state.isSubmitPromptOpen}
              onLeave={this.onModalLeave}
              message='Your blogpost has been saved successfully.'
          />
        </>
    );
  }
}

// Custom overrides for "code" style.
const styleMap = {
  CODE: {
    backgroundColor: 'rgba(0, 0, 0, 0.05)',
    fontFamily: '"Inconsolata", "Menlo", "Consolas", monospace',
    fontSize: 16,
    padding: 2,
  },
};

function getBlockStyle(block) {
  switch (block.getType()) {
  case 'blockquote': return 'RichEditor-blockquote';
  default: return null;
  }
}

class StyleButton extends React.Component {
  constructor() {
    super();
    this.onToggle = (e) => {
      e.preventDefault();
      this.props.onToggle(this.props.style);
    };
  }

  render() {
    let className = s['RichEditor-styleButton'];
    if (this.props.active) {
      className += ' ' + s['RichEditor-activeButton'];
    }

    return (
      <span className={className} onMouseDown={this.onToggle}>
        {this.props.label}
      </span>
    );
  }
}

const BLOCK_TYPES = [
  {label: 'H1', style: 'header-one'},
  {label: 'H2', style: 'header-two'},
  {label: 'H3', style: 'header-three'},
  {label: 'H4', style: 'header-four'},
  {label: 'H5', style: 'header-five'},
  {label: 'H6', style: 'header-six'},
  {label: 'Blockquote', style: 'blockquote'},
  {label: 'UL', style: 'unordered-list-item'},
  {label: 'OL', style: 'ordered-list-item'},
  {label: 'Code Block', style: 'code-block'},
];

const BlockStyleControls = (props) => {
  const {editorState} = props;
  const selection = editorState.getSelection();
  const blockType = editorState
        .getCurrentContent()
        .getBlockForKey(selection.getStartKey())
        .getType();

  return (
    <div className={s['RichEditor-controls']}>
       {BLOCK_TYPES.map((type) =>
                       <StyleButton
                         key={type.label}
                         active={type.style === blockType}
                         label={type.label}
                         onToggle={props.onToggle}
                         style={type.style}
                       />
                      )}
    </div>
  );
};

var INLINE_STYLES = [
  {label: 'Bold', style: 'BOLD'},
  {label: 'Italic', style: 'ITALIC'},
  {label: 'Underline', style: 'UNDERLINE'},
  {label: 'Monospace', style: 'CODE'},
];

const InlineStyleControls = (props) => {
  const currentStyle = props.editorState.getCurrentInlineStyle();

  return (
    <div className="RichEditor-controls">
       {INLINE_STYLES.map((type) =>
                         <StyleButton
                           key={type.label}
                           active={currentStyle.has(type.style)}
                           label={type.label}
                           onToggle={props.onToggle}
                           style={type.style}
                         />
                        )}
    </div>
  );
};


export default withRouter(RichTextEditor);



