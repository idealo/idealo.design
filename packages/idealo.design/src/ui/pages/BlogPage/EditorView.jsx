import React from 'react'

import Editor from './Editor'
import requiredAuthentication from "./requiredAuthentication";

function EditorView({}) {
  return  <Editor />
}

// console.log('requiredAuthentication',requiredAuthentication);

export default EditorView;