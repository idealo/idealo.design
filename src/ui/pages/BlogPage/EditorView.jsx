import React from "react";

import Editor from "./Editor";
import requiredAuthentication from "../requiredAuthentication";

function EditorView() {
  return <Editor />;
}

export default requiredAuthentication(EditorView);
