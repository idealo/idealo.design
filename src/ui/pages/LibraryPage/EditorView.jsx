import React from "react"
import Editor from "./Editor/Editor"
import requiredAuthentication from "../requiredAuthentication";

function EditorView(){
    return <Editor/>
}

export default requiredAuthentication(EditorView)