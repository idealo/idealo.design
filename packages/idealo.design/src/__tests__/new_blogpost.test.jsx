import React from 'react';
import { create } from "react-test-renderer"
import { RichTextEditor } from '../ui/pages/BlogPage/Editor/Editor'
import EditorView from "../ui/pages/BlogPage/EditorView";


// EditorView can't be accessed by an unauthenticated user
it('logged out user can not access editor', () => {
    const editorTest = create(<EditorView />);
    expect(editorTest.toJSON()).toMatchSnapshot();
});

// RichTextEditor gets rendered correctly and the editor is empty
it('logged out user can not access editor', done => {
    try{
        const RichTextEditorProps = {
            match: {
                params: ''
            }
        };
        const editorTest = create(<RichTextEditor {...RichTextEditorProps} />);
        editorTest.getInstance();
        expect(editorTest.toJSON()).toMatchSnapshot();
        done();
    }catch (error){
        done(error)
    }
});

