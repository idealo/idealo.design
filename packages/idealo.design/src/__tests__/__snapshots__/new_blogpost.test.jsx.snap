// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logged out user can not access editor 1`] = `
<div>
  <div>
    Please login in order to view this part of the application.
  </div>
</div>
`;

exports[`rich text editor gets rendered correctly 1`] = `
Array [
  <h1>
    Create blogpost
  </h1>,
  <div
    className="InputFields"
  >
    <input
      className=""
      name="title"
      onChange={[Function]}
      placeholder="Titel"
      value=""
    />
    <form
      className="select-container"
      name="category"
    >
      <div
        className=" css-2b097c-container"
        onKeyDown={[Function]}
      >
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className=" css-yk16xz-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" css-g1d714-ValueContainer"
          >
            <div
              className=" css-1uccc91-singleValue"
            >
              
            </div>
            <div
              className="css-b8ldur-Input"
            >
              <div
                className=""
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-autocomplete="list"
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  disabled={false}
                  id="react-select-2-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  spellCheck="false"
                  style={
                    Object {
                      "background": 0,
                      "border": 0,
                      "boxSizing": "content-box",
                      "color": "inherit",
                      "fontSize": "inherit",
                      "label": "input",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "1px",
                    }
                  }
                  tabIndex="0"
                  type="text"
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
          </div>
          <div
            className=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              className=" css-tlfecz-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                aria-hidden="true"
                className="css-tj5bde-Svg"
                focusable="false"
                height={20}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>,
  <div
    className="RichEditor-root"
  >
    <div
      className="RichEditor-controls"
    >
      <span
        className="RichEditor-styleButton"
        onMouseDown={[Function]}
      >
        H1
      </span>
      <span
        className="RichEditor-styleButton"
        onMouseDown={[Function]}
      >
        H2
      </span>
      <span
        className="RichEditor-styleButton"
        onMouseDown={[Function]}
      >
        H3
      </span>
      <span
        className="RichEditor-styleButton"
        onMouseDown={[Function]}
      >
        H4
      </span>
      <span
        className="RichEditor-styleButton"
        onMouseDown={[Function]}
      >
        H5
      </span>
      <span
        className="RichEditor-styleButton"
        onMouseDown={[Function]}
      >
        H6
      </span>
      <span
        className="RichEditor-styleButton"
        onMouseDown={[Function]}
      >
        Blockquote
      </span>
      <span
        className="RichEditor-styleButton"
        onMouseDown={[Function]}
      >
        UL
      </span>
      <span
        className="RichEditor-styleButton"
        onMouseDown={[Function]}
      >
        OL
      </span>
      <span
        className="RichEditor-styleButton"
        onMouseDown={[Function]}
      >
        Code Block
      </span>
    </div>
    <div
      className="RichEditor-controls"
    >
      <span
        className="RichEditor-styleButton"
        onMouseDown={[Function]}
      >
        Bold
      </span>
      <span
        className="RichEditor-styleButton"
        onMouseDown={[Function]}
      >
        Italic
      </span>
      <span
        className="RichEditor-styleButton"
        onMouseDown={[Function]}
      >
        Underline
      </span>
      <span
        className="RichEditor-styleButton"
        onMouseDown={[Function]}
      >
        Monospace
      </span>
    </div>
    <div
      onClick={[Function]}
    >
      <div
        className="DraftEditor-root"
      >
        <div
          className="public-DraftEditorPlaceholder-root"
        >
          <div
            className="public-DraftEditorPlaceholder-inner"
            id="placeholder-4bacg"
            style={
              Object {
                "whiteSpace": "pre-wrap",
              }
            }
          >
            Tell a story...
          </div>
        </div>
        <div
          className="DraftEditor-editorContainer"
        >
          <div
            aria-describedby="placeholder-4bacg"
            aria-expanded={null}
            className="notranslate public-DraftEditor-content"
            contentEditable={true}
            onBeforeInput={[Function]}
            onBlur={[Function]}
            onCompositionEnd={[Function]}
            onCompositionStart={[Function]}
            onCopy={[Function]}
            onCut={[Function]}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            onInput={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            onMouseUp={[Function]}
            onPaste={[Function]}
            onSelect={[Function]}
            role="textbox"
            spellCheck={true}
            style={
              Object {
                "WebkitUserSelect": "text",
                "outline": "none",
                "userSelect": "text",
                "whiteSpace": "pre-wrap",
                "wordWrap": "break-word",
              }
            }
            suppressContentEditableWarning={true}
          >
            <div
              data-contents="true"
            >
              <div
                className={null}
                data-block={true}
                data-editor="4bacg"
                data-offset-key="c313h-0-0"
              >
                <div
                  className="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                  data-offset-key="c313h-0-0"
                >
                  <span
                    data-offset-key="c313h-0-0"
                    style={Object {}}
                  >
                    <br
                      data-text="true"
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="newBlogPostButtons"
  >
    <button
      className="SubmitButton"
      onClick={[Function]}
    >
      Submit
    </button>
    <button
      className="CancelButton"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>,
]
`;
